benchmark/url/url-searchparams-iteration.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const common = require('../common.js');
+const URLSearchParams = require('url').URLSearchParams;
+const v8 = require('v8');
+
+v8.setFlagsFromString('--allow_natives_syntax');
+
+const bench = common.createBenchmark(main, {
+  method: ['forEach', 'iterator'],
+  n: [1e6]
+});
+
+const str = 'one=single&two=first&three=first&two=2nd&three=2nd&three=3rd';
+
+function forEach(n) {
+  const params = new URLSearchParams(str);
+  const noop = () => {};
+  eval('%OptimizeFunctionOnNextCall(params.forEach)');
```

Comment:
To make `OptimizeFunctionOnNextCall` work properly, we should call it **after**  making the first call to `params.forEach`(it needs enough type info to optimize, which can be gathered in the first call).

BTW I am not sure we do need to do this step, because the optimizing compiler doesn't necessarily have enough type feedback to properly optimize this with just one call. I believe this conventions is to prevent the OSR(on-stack replacement) from affecting the result, but if the compiler doesn't have enough type info during the first optimization, it might still kick in again in the loop.